src/nvim/api/vim.c

```diff
@@ -2450,3 +2451,191 @@ void nvim_del_user_command(String name, Error *err)
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)USER_CMD(ea.useridx)->uc_name));
+  } else if (ea.cmdidx == CMD_USER_BUF) {
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)USER_CMD_GA(&curbuf->b_ucmds, ea.useridx)->uc_name));
+  } else {
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)get_command_name(NULL, ea.cmdidx)));
+  }
+  PUT(result, "line1", INTEGER_OBJ(ea.line1));
+  PUT(result, "line2", INTEGER_OBJ(ea.line2));
+  PUT(result, "force", BOOLEAN_OBJ(ea.forceit));
+  PUT(result, "args", ARRAY_OBJ(args));
+
+  if (ea.argt & EX_EXTRA) {
+    if (ea.argt & EX_NOSPC) {
+      if (ea.argt & EX_NEEDARG) {
+        nargs[0] = '1';
+      } else {
+        nargs[0] = '?';
+      }
+    } else if (ea.argt & EX_NEEDARG) {
+      nargs[0] = '+';
+    } else {
+      nargs[0] = '*';
+    }
+  } else {
+    nargs[0] = '0';
+  }
+  nargs[1] = '\0';
+  PUT(result, "nargs", CSTR_TO_OBJ(nargs));
+
+  switch (ea.addr_type) {
+  case ADDR_LINES:
+    addr = "line";
+    break;
+  case ADDR_ARGUMENTS:
+    addr = "arg";
+    break;
+  case ADDR_BUFFERS:
+    addr = "buf";
+    break;
+  case ADDR_LOADED_BUFFERS:
+    addr = "load";
+    break;
+  case ADDR_WINDOWS:
+    addr = "win";
+    break;
+  case ADDR_TABS:
+    addr = "tab";
+    break;
+  case ADDR_QUICKFIX:
+    addr = "qf";
+    break;
+  case ADDR_NONE:
+    addr = "none";
+    break;
+  default:
+    addr = "?";
+    break;
+  }
+  PUT(result, "addr", CSTR_TO_OBJ(addr));
+  PUT(result, "nextcmd", CSTR_TO_OBJ((char *)ea.nextcmd));
+
+  Dictionary mods = ARRAY_DICT_INIT;
+  PUT(mods, "silent", BOOLEAN_OBJ(cmdinfo.silent));
+  PUT(mods, "emsg_silent", BOOLEAN_OBJ(cmdinfo.emsg_silent));
+  PUT(mods, "sandbox", BOOLEAN_OBJ(cmdinfo.sandbox));
+  PUT(mods, "noautocmd", BOOLEAN_OBJ(cmdinfo.noautocmd));
+  PUT(mods, "tab", INTEGER_OBJ(cmdmod.tab));
+  PUT(mods, "verbose", INTEGER_OBJ(cmdinfo.verbose));
+  PUT(mods, "browse", BOOLEAN_OBJ(cmdmod.browse));
+  PUT(mods, "confirm", BOOLEAN_OBJ(cmdmod.confirm));
+  PUT(mods, "hide", BOOLEAN_OBJ(cmdmod.hide));
+  PUT(mods, "keepalt", BOOLEAN_OBJ(cmdmod.keepalt));
+  PUT(mods, "keepjumps", BOOLEAN_OBJ(cmdmod.keepjumps));
+  PUT(mods, "keepmarks", BOOLEAN_OBJ(cmdmod.keepmarks));
+  PUT(mods, "keeppatterns", BOOLEAN_OBJ(cmdmod.keeppatterns));
+  PUT(mods, "lockmarks", BOOLEAN_OBJ(cmdmod.lockmarks));
+  PUT(mods, "noswapfile", BOOLEAN_OBJ(cmdmod.noswapfile));
+
+  Dictionary splitmods = ARRAY_DICT_INIT;
+  PUT(splitmods, "vertical", BOOLEAN_OBJ(cmdmod.split & WSP_VERT));
+  PUT(splitmods, "aboveleft", BOOLEAN_OBJ(cmdmod.split & WSP_ABOVE));
+  PUT(splitmods, "belowright", BOOLEAN_OBJ(cmdmod.split & WSP_BELOW));
+  PUT(splitmods, "topleft", BOOLEAN_OBJ(cmdmod.split & WSP_TOP));
+  PUT(splitmods, "botright", BOOLEAN_OBJ(cmdmod.split & WSP_BOT));
+
+  PUT(mods, "split", DICTIONARY_OBJ(splitmods));
+  PUT(result, "mods", DICTIONARY_OBJ(mods));
+
+  Array magic = ARRAY_DICT_INIT;
+  if (cmdinfo.magic.percent) {
+    ADD(magic, STRING_OBJ(STATIC_CSTR_TO_STRING("%")));
```

Comment:
Just for reference: another one is `*` and probably various wildcard characters.

Reply:
Looking into the code, it seems those are all related to `%` and are expanded when `%` is expanded, maybe I should put that in the docs.

Reply:
Alright it's in the docs now :)